packages/wdio-rpc/README.md

```diff
@@ -0,0 +1,266 @@
+- `workerEvent()` - Process worker events
+- `workerResponse()` - Handle worker responses
+
+### Client Functions
+Functions exposed by the test environment that can be called by the main runner:
+
+- `consoleMessage()` - Handle console output
+- `commandResponseMessage()` - Process command responses
+- `hookResultMessage()` - Handle hook results
+- `expectResponseMessage()` - Process expectation responses
+- `coverageMap()` - Handle coverage data
+- `runCommand()` - Execute commands
+- `triggerHook()` - Trigger test hooks
+- `expectRequest()` - Handle expectation requests
+- `expectMatchersRequest()` - Process matcher requests
+- `browserTestResult()` - Handle browser test results
+
+## Usage
+
+### Basic Setup
+
+```typescript
+import { createClientRpc, createServerRpc } from '@wdio/rpc'
+
+// Create server RPC instance
+const serverRpc = createServerRpc({
+    sessionStarted: (data) => {
+        console.log('Session started:', data.content.sessionId)
+    },
+    sessionEnded: (data) => {
+        console.log('Session ended:', data.cid)
+    }
+})
+
+// Create client RPC instance
+const clientRpc = createClientRpc({
+    consoleMessage: (data) => {
+        console.log('Console:', data.args)
+    },
+    runCommand: async (data) => {
+        // Execute command and return result
+        return 'command result'
+    }
+})
+```
+
+### With Error Handling
+
+```typescript
+import { createClientRpc } from '@wdio/rpc'
+
+const clientRpc = createClientRpc(
+    {
+        consoleMessage: (data) => {
+            console.log('Console:', data.args)
+        }
+    },
+    {
+        onError: (error) => {
+            console.error('RPC Error:', error.message)
+            // Handle error appropriately
+        },
+        timeout: 5000,
+        retries: 3
+    }
+)
+```
+
+### Integration with WebdriverIO
+
+The RPC system is automatically integrated into WebdriverIO's architecture:
+
+#### Browser Runner Integration
+```typescript
+// In browser-runner communicator
+import { createServerRpc } from '@wdio/rpc'
+
+const communicator = new ServerWorkerCommunicator(config)
+communicator.register(server, worker)
+
+// RPC handlers are automatically set up for:
+// - Session management
+// - Worker communication
+// - Coverage collection
+// - Custom commands
+```
+
+#### Test Runner Integration
+```typescript
+// In test runner
+import { createClientRpc } from '@wdio/rpc'
+
+const framework = new BrowserFramework(cid, config, specs, reporter)
+// RPC is automatically configured for:
+// - Hook execution
+// - Command execution
+// - Expectation handling
+// - Console message handling
+```
+
+## Message Flow
+
+```
+┌─────────────────┐    RPC Messages    ┌─────────────────┐
+│   Test Runner   │ ◄────────────────► │  Browser Runner │
+│   (Server)      │                    │    (Client)     │
+└─────────────────┘                    └─────────────────┘
+         │                                       │
+         │                                       │
```

Comment:
Can we label these communications as well?
```suggestion
         │  Node.js RPC                │ ? (not sure here)
```

Also can we reference which files implement which of these boxes (except for browser environment)?